src/org/labkey/test/pages/assay/ChooseAssayTypePage.java

```diff
@@ -30,12 +27,33 @@ public static ChooseAssayTypePage beginAt(WebDriverWrapper webDriverWrapper, Str
         return new ChooseAssayTypePage(webDriverWrapper.getDriver());
     }
 
+    @Override
+    protected void waitForPage()
+    {
+        waitFor(() -> {
+            try
+            {
+                return elementCache().stdAssayTabLocator().existsIn(getDriver()) &&
+                        elementCache().specialtyAssayTabLocator().existsIn(getDriver());
+            }
+            catch (NoSuchElementException retry)
+            {
+                return false;
+            }
```

Comment:
There's no realistic way for this to throw `NoSuchElementException`. The purpose of `existsIn` is to handle missing elements.
```suggestion
            return elementCache().stdAssayTabLocator().existsIn(getDriver()) &&
                    elementCache().specialtyAssayTabLocator().existsIn(getDriver());
```

Reply:
Alternatively, you could use the element cache to avoid finding the tabs repeatedly. (`LazyWebElement.isDisplayed` handles non-existent elements as well).
```suggestion
            return elementCache().stdAssayTab.isDisplayed() &&
                    elementCache().specialtyAssayTab.isDisplayed();
```

Reply:
Thanks!